cmd/signature-v4-utils.go

```diff
@@ -157,3 +157,25 @@ func extractSignedHeaders(signedHeaders []string, reqHeaders http.Header) (http.
 
 	return extractedSignedHeaders, ErrNone
 }
+
+// Sequential spaces regular expression
+var seqSpaces = regexp.MustCompile("  +")
+
+// Trim leading and trailing spaces and replace sequential spaces with one space, following Trimall()
+// in http://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+func sigV4TrimAll(input string) string {
```

Comment:
the code in object-handlers_test.go and test-utils_test.go uses SigV4 though: malformChunkSizeSigV4() malformDataSigV4() truncateChunkByHalfSigv4()